contracts/Vault.sol

```diff
@@ -123,16 +146,29 @@ contract Vault is
     function deposit(
         uint256 _amount,
         address _recipient
-    ) external override returns (uint256) {
-        return _deposit(_amount, _recipient);
+    ) public override returns (uint256) {
+        if (emergencyShutdown) revert Vault__V11();
+        uint256 shares = _issueSharesForAmount(_recipient, _amount);
+        token.safeTransferFrom(msg.sender, address(this), _amount);
+        return shares;
     }
 
     function withdraw(
         uint256 _maxShares,
         address _recipient,
         uint256 _maxLoss
-    ) external override {
-        _initiateWithdraw(_maxShares, _recipient, _maxLoss);
+    ) public override WithdrawInProgress {
+        _transfer(msg.sender, address(this), _maxShares);
```

Comment:
Reentrant??

Reply:
I don't know what to do with it, cause of space, is there any protection in WithdrawInProgress modifier in case of reentrant attack?